packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts

```diff
@@ -127,28 +143,48 @@ describe('WorkspaceContext', () => {
     let createConnectionStub: jest.SpyInstance;
 
     beforeEach(() => {
-      createAuthStub = jest
-        .spyOn(AuthInfo, 'create');
-      createConnectionStub = jest
-        .spyOn(Connection, 'create');
+      createAuthStub = jest.spyOn(AuthInfo, 'create');
+      createConnectionStub = jest.spyOn(Connection, 'create');
     });
 
     it('should return connection for the default org', async () => {
       authInfoMock.create.mockResolvedValue(mockAuthInfo as any);
-      authInfoMock.create.mockResolvedValue(mockAuthInfo as any);
-      connectionMock.create.mockResolvedValue(mockConnection as unknown as Promise<Connection<any>>);
+      connectionMock.create.mockResolvedValue(
+        (mockConnection as unknown) as Promise<Connection<any>>
+      );
       const connection = await workspaceContextUtil.getConnection();
-      expect(connectionMock.create).toHaveBeenCalledWith({ authInfo: mockAuthInfo });
+      expect(connectionMock.create).toHaveBeenCalledWith({
+        authInfo: mockAuthInfo
+      });
       expect(connection).toEqual(mockConnection);
     });
 
     it('should return a cached connection for the default org if there is one', async () => {
       authInfoMock.create.mockResolvedValue(mockAuthInfo as any);
-      connectionMock.create.mockResolvedValue(mockConnection as unknown as Promise<Connection<any>>);
+      connectionMock.create.mockResolvedValue(
+        (mockConnection as unknown) as Promise<Connection<any>>
+      );
       await workspaceContextUtil.getConnection();
       await workspaceContextUtil.getConnection();
 
       expect(connectionMock.create).toHaveBeenCalledTimes(1);
     });
+
+    it('should not throw error if there is a username set', async () => {
+      const connection = await workspaceContextUtil.getConnection();
+      expect(() => connection).not.toThrow();
+    });
+
+    it('should throw error if there is no username set', async () => {
+      getUsernameStub.mockReturnValue(undefined);
+
+      await mockWatcher.onDidChange.mock.calls[0][0]();
+
+      const message = nls.localize('error_no_default_username');
+      // tslint:disable-next-line:no-floating-promises
+      expect(async () => {
+        await workspaceContextUtil.getConnection();
+      }).rejects.toThrowError(message);
```

Comment:
Every time I do this I have to explore a little to find the right combo to get the expect to work.  Also the tslint exclusion is lame, but it's about time to pull the trigger on just moving everything to eslint so I'm ok with it.